### DISCUSSION TITLE
Removal of potentially biological differences across batches

### DISCUSSION THREAD

**Discussion by Troylimyj** [ImmuneDynamics/Spectre#22] [! News]
Dear Tom, 

Thank you for your excellent work in developing this very useful package. I read with interest your normalization strategy involving scaling of each marker to a specific percentile followed by quantile normalisation with cytoNorm.

My question is, would this risk increasing the chances of removing potentially biological difference across batches, particularly as each sample is essentially normalised twice?

Best wishes
Troy

**Comment by tomashhurst** [comment]
Hi @Troylimyj ,

That's an excellent question! If using the _reference_ controls (i.e. biologically similar samples that are acquired with each batch) to learn the batch effects, any biological differences between individual samples should be well preserved. The coarse alignment is really just an intermediate step to make sure that cells in the reference controls across batches will end up being assigned to the same metaclusters, because once the cells have been assigned, the 'fine' alignment should take care of the rest.

Am important note though, is that we can only really adjust for markers that are _represented_ by the reference controls. For example, on PBMCs, CD4/CD8 etc could be adjusted fairly well, but various forms of activation or potentially exhaustion markers might not be expressed on normal/healthy PBMCs being used for reference controls. In this case we would _not_ adjust those markers, but rather investigate them as is (but they can be examined much more easily on clusters, as the main phenotyping markers should have been aligned well).

Hopefully that makes sense. We've got an update coming to that workflow that essentially splits it into two workflows: one for straight-forward batch alignment which uses _just_ CytoNorm, and then another that handles more complex integration (different panels, technologies etc) which should clean up the logic a little.

Tom
